metabase/metabase
# Fix an error during updating cards which are dashcards

Closes https://github.com/metabase/metabase/issues/31766

### Description

Dashboards with question and text/link cards corrupted dashcards data during saving updates. It turned out that UI cards without a reference to actual card weren't filter during data restoring.

### How to verify

1. Create a dashboard with a question and text card, save it. 
2. Edit text card on the dashboard and save it again.
3. Drill down to the question from dashboard, change visualization type, click save and replace.
There shouldn't be an error like in the issue and question should be saved.

### Demo

https://www.loom.com/share/a01a50f859074500971b0bdb2c70b2e3

### Checklist
